subsys/metrics/src/main/java/org/commonjava/indy/metrics/zabbix/cache/MetricsZabbixCache.java

```diff
@@ -0,0 +1,122 @@
+package org.commonjava.indy.metrics.zabbix.cache;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.commonjava.indy.metrics.zabbix.cache.event.MetricsZabbixCacheEventManager;
+import org.commonjava.indy.subsys.datafile.DataFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Created by xiabai on 5/5/17.
+ */
+@ApplicationScoped
+public class MetricsZabbixCache
+{
```

Comment:
I'd make this implement StartupAction, then @Inject ZabbixInit (which I'd rename to something like ZabbixCacheStorage) and call methods in that component to load the data from file. When values change in this class, I wouldn't mess with events. I'd just call the ZabbixCacheStorage / ZabbixInit component directly to write the relevant map to disk.